### ISSUE TITLE
Content Density Service Improvement - Composition

### ISSUE THREAD

**Issue by JKMarkowski** [SAP/fundamental-ngx#5041]
Content Density usage inside core components should be improved.
One of composition features should be used for every single component, to get rid of redundant variables.
This will provide reusability for repetitive behaviours, because such will be in only one place.

IMO this could be done by creating base class with properties, that are included into every component.

- [ ] include  https://github.com/SAP/fundamental-ngx/issues/5689
- [ ] include https://github.com/SAP/fundamental-ngx/issues/7108
- [ ] review the PR for https://github.com/SAP/fundamental-ngx/issues/7765 and refactor if needed

**Comment by droshev** [comment]
@g-cheishvili Could you summarize your proposal for the content density refactoring here?

**Comment by g-cheishvili** [comment]
@droshev 
Goal: 
* Have centralised place for managing content density configuration - Done
* Have local configuration for certain elements in application
* Get rid of unnecessary variables across multiple components
* Have ability to store selected configuration out of the box
* Read content density configuration parameter from URL and preferably configurable name for it - Partially done

Implementation proposal:
Implement directives `[fdCompact]="true | false"` and `[fdCondensed]="true | false"` for managing locally enforced configuration. Value, provided to these variables will override globally selected one and on top of it, any component under these directives will detect local enforcement and will use it.
Implement interface, for storing selected configuration, something along the lines
```ts
interface ContentDensityStorage {
    get: () => Observable<'condensed' | 'cozy' | 'compact'>;
    set: (val: 'condensed' | 'cozy' | 'compact') => Observable<void>
}
```
This way, storage is up to the user's preference, we can implement three variants out of the box, `LocalContentDensityStorage`, `MemoryContentDensityStorage`, `UrlContentDensityStorage` and then add one generic implementation, something like `HttpContentDensityStorage`, which will be extendable class, with endpoints provided by user. Also all of the above out of the box implementations will have their own DI deps params, which will be configured by the user. Plus all of these functions can, and should be SSR compatible.
We should get rid of the inputs that are added currently to the components across the library, things such as `[cozy]`, or `[compact]` and so on. They should all be excluded from components themselves and replaced with one provider, which will be responsible for detecting global and local configs, applying them to host components.
Also for ease of development down the road, we should create decorator, something like
```
@SupportsCompact({
    compactClassName: string
    ...Some additional configs, that will be visible during development
})
```
This will just add required properties to the class and along with the up mentioned provider will make changes to the host and child components, depending on requirements
